solarforecastarbiter/metrics/event.py

```diff
@@ -0,0 +1,230 @@
+"""Event forecast error metrics."""
+
+import numpy as np
+
+
+def _event2count(obs, fx):
+    """Convert events (True/False) into counts.
+
+    Given forecasts and observations of events (True=event occurred,
+    False=event did not occur), the pairs of forecasts and observations can be
+    placed into four categories:
+    - True Positive (TP): forecast = event, observed = event
+    - False Positive (FP): forecast = event, observed = no event
+    - True Negative (TN): forecast = no event, observed = no event
+    - False Negative (FN): forecast = no event, observed = event
+
+    Parameters
+    ----------
+    obs : (n,) array-like
+        Observed event values (True=event, False=no event).
+    fx : (n,) array-like
+        Forecasted event values (True=event, False=no event).
+
+    Returns
+    -------
+    tp : int
+        Number of true positives.
+    fp : int
+        Number of false positives.
+    tn : int
+        Number of true negatives.
+    fn : int
+        Number of false negatives.
+
+    Raises
+    ------
+    RuntimeError
+        If there is no forecast or observation timeseries data.
+
+    """
+
+    if isinstance(obs, (list, tuple)):
+        obs = np.array(obs)
+    if isinstance(fx, (list, tuple)):
+        fx = np.array(fx)
+
+    if len(obs) == 0:
+        raise RuntimeError("No Observation timeseries data.")
+    elif len(fx) == 0:
+        raise RuntimeError("No Forecast timeseries data.")
+
```

Comment:
should we also check for length mismatch?

```python
In [3]: fx = np.array([True])

In [4]: obs = np.array([True, True])

In [5]: np.count_nonzero(np.logical_and(fx, obs))
Out[5]: 2
```

Reply:
Good point. I wasn't sure whether that's something we can assume gets handled by before (e.g. in `metrics.preprocessing`), but it's easy enough to add an explicit check here as well.

Reply:
I think it's worthwhile here because the failure mode is especially dangerous.

Reply:
The new version of this file now has an explicit check that raises a `RuntimeError` if `len(obs) != len(fx)`.